spyder_kernels/console/kernel.py

```diff
@@ -252,10 +252,11 @@ def load_data(self, filename, ext):
         if error_message:
             return error_message
 
-        for key in list(data.keys()):
-            new_key = fix_reference_name(key, blacklist=list(glbs.keys()))
-            if new_key != key:
-                data[new_key] = data.pop(key)
+        if not override:
+            for key in list(data.keys()):
+                new_key = fix_reference_name(key, blacklist=list(glbs.keys()))
```

Comment:
Here we do need to keep the list() since keys() on newer python versions does not return a list.